nilearn/decomposition/tests/test_canica.py

```diff
@@ -2,220 +2,66 @@
-        """Check compliance with sklearn estimators."""
-        check(estimator)
-
-    @pytest.mark.xfail(reason="invalid checks should fail")
-    @pytest.mark.parametrize(
-        "estimator, check, name",
-        check_estimator(estimators=ESTIMATORS_TO_CHECK, valid=False),
-    )
-    def test_check_estimator_sklearn_invalid(estimator, check, name):  # noqa: ARG001
-        """Check compliance with sklearn estimators."""
-        check(estimator)
-
-else:
-
-    @parametrize_with_checks(
-        estimators=ESTIMATORS_TO_CHECK,
-        expected_failed_checks=return_expected_failed_checks,
-    )
-    def test_check_estimator_sklearn(estimator, check):
-        """Check compliance with sklearn estimators."""
-        check(estimator)
-
-
-@pytest.mark.parametrize(
-    "estimator, check, name",
-    nilearn_check_estimator(estimators=ESTIMATORS_TO_CHECK),
+from nilearn.decomposition.tests.conftest import (
+    RANDOM_STATE,
+    check_decomposition_estimator,
 )
-def test_check_estimator_nilearn(estimator, check, name):  # noqa: ARG001
-    """Check compliance with nilearn estimators rules."""
-    check(estimator)
+from nilearn.image import get_data, iter_img
+from nilearn.surface.surface import get_data as get_surface_data
 
 
-def test_threshold_bound_error(canica_data):
+@pytest.mark.parametrize("data_type", ["nifti"])
+def test_threshold_bound_error(canica_data_single_img):
     """Test that an error is raised when the threshold is higher \
     than the number of components.
     """
     with pytest.raises(ValueError, match="Threshold must not be higher"):
         canica = CanICA(n_components=4, threshold=5.0)
-        canica.fit(canica_data)
-
-
-def test_transform_and_fit_errors(mask_img):
-    canica = CanICA(mask=mask_img, n_components=3)
+        canica.fit(canica_data_single_img)
 
-    # error when empty list of provided.
-    with pytest.raises(
-        ValueError,
-        match="Need one or more Niimg-like objects as input, "
-        "an empty list was given.",
-    ):
-        canica.fit([])
 
-    # error is raised when no data is passed.
-    with pytest.raises(
-        TypeError,
-        match="missing 1 required positional argument: 'imgs'",
-    ):
-        canica.fit()
-
-
-def test_percentile_range(rng, canica_data):
+@pytest.mark.parametrize("data_type", ["nifti"])
+def test_percentile_range(rng, canica_data_single_img):
     """Test that a warning is given when thresholds are stressed."""
     edge_case = rng.integers(low=1, high=10)
 
     # stress thresholding via edge case
     canica = CanICA(n_components=edge_case, threshold=float(edge_case))
 
     with pytest.warns(UserWarning, match="obtained a critical threshold"):
-        canica.fit(canica_data)
+        canica.fit(canica_data_single_img)
 
 
-def test_canica_square_img(mask_img):
-    data, components, rng = _make_canica_test_data(n_subjects=8)
-
+@pytest.mark.parametrize("data_type", ["nifti"])
```

Comment:
and here

Reply:
for this one it seems that the mock data has been tailored for this one to pass with volume data: make it pass for surface data is a headache

so I am not testing it with surface yet

could be done in a later PR